packages/backend/src/services/AsyncQueryService/AsyncQueryService.ts

```diff
@@ -165,6 +183,164 @@ export class AsyncQueryService extends ProjectService {
         });
     }
 
+    private getCacheExpiresAt(baseDate: Date) {
+        return new Date(
+            baseDate.getTime() +
+                this.lightdashConfig.results.cacheStateTimeSeconds * 1000,
+        );
+    }
+
+    async createOrGetExistingCache(
```

Comment:
Moved all the methods from Cache service to this service. The only one I changed the logic is this one. 

It calls the cache service to verify there is a cache match. 
Otherwise, upserts the cache row to avoid conflict with existing row with the same cache_key.